pilot/pkg/bootstrap/server.go

```diff
@@ -825,11 +825,14 @@ func (s *Server) initSecureGrpcListener(args *PilotArgs) error {
 		return fmt.Errorf("invalid port(%s) in ISTIOD_ADDR(%s): %v", port, istiodAddr, err)
 	}
 
-	// Create DNS certificates. This allows injector, validation to work without Citadel, and
-	// allows secure SDS connections to Istiod.
-	err = s.initDNSCerts(host, features.IstiodServiceCustomHost.Get(), args.Namespace)
-	if err != nil {
-		return err
+	// Generate DNS certificates only if custom certs are not provided via args.
+	if args.TLSOptions.CaCertFile == "" && args.TLSOptions.CertFile == "" && args.TLSOptions.KeyFile == "" {
```

Comment:
This might be too strict - if key and cert are not empty. 

We want to generate own certs even if CaCertFile is provided - unless we have another way to 
specify additional CA roots (for MC). I don't remember all the options in this area, but ideally 
we wouldn't have too many duplicated ways to configure that...

Reply:
I mean: I would only check key, cert not empty. 

For CaCertFile - we should unify the options, so there is only one way to specify the set of CAs that are trusted. AFAIK this is not even used when starting the TLS server.